lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs

```diff
@@ -1610,7 +1598,7 @@ coinSelectionErrorToBalanceTxError era = withConstraints era $ \case
         ErrBalanceTxInsufficientCollateral
         ErrBalanceTxInsufficientCollateralError
             { largestCombinationAvailable
-                = largestCombinationAvailable
+                = withConstraints era $ largestCombinationAvailable
                 & fmap W.TokenBundle.fromCoin
                 & toExternalUTxOMap
                 & fromWalletUTxO era
```

Comment:
Suggestion: avoid mixing `$` and `&`: mixing these means that the code can no longer be read purely from left-to-right or from right-to-left.

I've made a fix for this here: e009054135ab30b555e08ba05c5f7f5a7588a351.

Reply:
However, this `withConstraints` should now be avoidable... (5afae154cfb544abe1714db1be85d56b31ba8337) I'll see what other `withConstraints` calls I can remove...